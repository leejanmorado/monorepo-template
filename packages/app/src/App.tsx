
import { Button } from '@components';
import { useState } from 'react';

const App: React.FC = () => {
	const [count, setCount] = useState<number>(0);

	const handleClick = () => {
		setCount(count + 1);
	};

	return (
		<div className='h-screen bg-gray-900 text-2xl sm:text-5xl lg:text-8xl'>
			<div className='flex h-full flex-col items-center justify-center'>
				<p className='mb-4 w-full text-center text-white'>Count: {count}</p>
				<Button onClick={handleClick}>Click Me</Button>
			</div>
		</div>
	);
};

export default App;
