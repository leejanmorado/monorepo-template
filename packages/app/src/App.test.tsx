import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import App from './App';

describe('Button component', () => {
	let container: HTMLElement;

	beforeEach(() => {
		({ container } = render(<App />));
	});

	it('matches DOM snapshot', () => {
		expect(container.firstChild).toMatchSnapshot();
	});

	it('increments counter', async () => {
		// arrange
		const user = userEvent.setup();
		const button = screen.getByRole('button', { name: 'Click Me' });

		// act
		await user.click(button);

		// assert
		expect(screen.getByText('Count: 1')).toBeInTheDocument();
	});
});
