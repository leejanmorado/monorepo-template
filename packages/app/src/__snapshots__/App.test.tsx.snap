// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button component > matches DOM snapshot 1`] = `
<div
  class="h-screen bg-gray-900 text-2xl sm:text-5xl lg:text-8xl"
>
  <div
    class="flex h-full flex-col items-center justify-center"
  >
    <p
      class="mb-4 w-full text-center text-white"
    >
      Count: 
      0
    </p>
    <div
      class="h-10"
    >
      <button
        class="rounded-md bg-white px-4 py-2 text-base text-black transition-all hover:bg-gray-300 active:bg-gray-300 active:py-[6px] active:px-[14px] active:text-sm  undefined"
      >
        Click Me
      </button>
    </div>
  </div>
</div>
`;
